wordpress-mobile/WordPress-FluxC-Android
# Issue/5870 save inbox notes

### Description
Save inbox notes into Room database and expose an observable to the database state so we can observe its changes from WooCommerce app.

Note: This code is only used in WooCommerce Inbox that has not been released yet as is hidden under feature flag. So we have absolute freedom to apply any modifications to the DB

### Testing instructions
The best way to test this is to download the apk  from this [PR](https://github.com/woocommerce/woocommerce-android/pull/6122) and go to More Menu -> Inbox and check that inbox notes are displayed
